test/verify/check-connection

```diff
@@ -459,32 +459,58 @@ class TestConnection(MachineCase):
     def testSocket(self):
         m = self.machine
 
-        self.assertIn("systemctl", m.execute("cat /etc/issue.d/cockpit.issue"))
-        self.assertIn("systemctl", m.execute("cat /etc/motd.d/cockpit"))
-        self.assertNotIn("9090", m.execute("cat /etc/motd.d/cockpit"))
-        m.start_cockpit()
+        # non-admin user
+        m.execute("useradd user")
+
+        # enable no-password login for 'admin' and 'user'
+        m.execute("passwd -d admin")
+        m.execute("passwd -d user")
+        self.sed_file('$ a\\\nPermitEmptyPasswords yes', '/etc/ssh/sshd_config',
+                      'systemctl restart sshd.service')
+
+        def assertInOrNot(string, result, expected):
+            if expected:
+                self.assertIn(string, result)
+            else:
+                self.assertNotIn(string, result)
+
+        def checkMotdForUser(string, user, expected):
+            result = m.execute(f"ssh -o StrictHostKeyChecking=no -n {user}@localhost")
+            assertInOrNot(string, result, expected)
+
+        def checkMotdContent(string, expected=True):
+            # Needs https://github.com/linux-pam/linux-pam/pull/292 (or PAM 1.5.0)
+            pam_checks_perms = (m.image in ['fedora-33', 'fedora-testing'])
```

Comment:
I prefer explicitly enumerating broken OSes, i.e. inverting the list. That's less prone to forgetting an OS and silently skipping parts of the test, and it's obvious when we can remove the special case (when the list is empty).

Reply:
I'm not sure I agree, when the "broken" list is so long...

Reply:
Sure it is long, but it's better to have it long now and drive it to zero, than having to add *new* operating systems (like fedora-34 or rhel-8-5 or so) to it. Special cases/bugs should be about a bounded set of known cases, not be the "good" code path.

Reply:
OK :)